python/ray/serve/_private/test_utils.py

```diff
@@ -731,13 +734,18 @@ def get_application_urls(
             for low latency benchmarking.
         is_websocket: Whether the url should be served as a websocket.
         exclude_route_prefix: The route prefix to exclude from the application.
+        check_app_is_running: Whether to check if the application is running before getting the URL.
```

Comment:
it's strange to put this in a getter function. wherever you are using this, I would recommend separating out a wait for condition to wait for the app to be running